sdk/typescript/src/transactions/blockData/v2.ts

```diff
@@ -95,13 +98,17 @@ export const Argument = transform(
 	| { GasCoin: true }
 	| { Input: number; type?: 'pure' | 'object' }
 	| { Result: number }
-	| { NestedResult: [number, number] },
+	| { NestedResult: [number, number] }
+	| { IntentResult: number }
+	| { NestedIntentResult: [number, number] },
 	// Output
 	| { $kind: 'GasCoin'; GasCoin: true }
 	| { $kind: 'Input'; Input: number; type?: 'pure' }
 	| { $kind: 'Input'; Input: number; type?: 'object' }
 	| { $kind: 'Result'; Result: number }
 	| { $kind: 'NestedResult'; NestedResult: [number, number] }
+	| { $kind: 'IntentResult'; IntentResult: number }
+	| { $kind: 'NestedIntentResult'; NestedIntentResult: [number, number] }
```

Comment:
I think these might not be needed, I will probably just remove them and use normal Result and NestedResults unless we run into a usecase where there is a good reason to differentiate them